MultiLangProgGuide/MultiLangProgGuide.rst

```diff
@@ -2330,31 +2348,38 @@ Go Str2ZwrS()
 
     Str2ZwrS(zwr *BufferT) int
 
-In the buffer referenced by :code:`zwr`, the :code:`Str2ZwrS()` method
-wraps `ydb_str2zwr_s()`_ and returns the :code:`BufferT` string
-in `zwrite format`_, with a return code of :code:`C.YDB_OK`. In the
-event the buffer is not large enough, the return code is
-:code:`C.YDB_ERR_INVSTRLEN` and :code:`zwr.GetLenUsed()` returns the
-length required (in this case, the value returned by
-:code:`zwr.GetLenUsed()` will be greater than that returned by
-:code:`zwr.GetLenAlloc()` until corrected by application code.
-
+The method wraps `ydb_str2zwr_s()`_ to provide the string in `zwrite
+format`_.
+
+- If :code:`len_alloc` is not large enough, set :code:`len_used` to
+  the required length, and return with a return code of
+  :code:`C.YDB_ERR_INVSTRLEN`. In this case, :code:`len_used` will be
+  greater than :code:`len_alloc` until corrected by application code.
+- Otherwise, set the buffer referenced by :code:`buf_addr` to the
+  `zwrite format`_ string, set :code:`len_used` to the length, and
+  return with a return code of :code:`C.YDB_OK`.
+   
 Go Zwr2StrS()
 -------------
 
 .. code-block:: go
 
-    wr2StrS(str *BufferT) int
+    Zwr2StrS(str *BufferT) int
+
+This method wraps `ydb_zwr2str_s()`_ and is the inverse of `Go
+Str2ZwrS()`_.
 
-In the buffer referenced by :code:`str`, the :code:`Zwr2StrS()` method
-wraps `ydb_zwr2str_s()` and returns the :code:`BufferT` string
-converted from `zwrite format`_ to its original, unencoded format,
-with a return code of :code:`C.YDB_OK`. In the event the buffer is not
-large enough, the return code is :code:`C.YDB_ERR_INVSTRLEN` and
-:code:`str.GetLenUsed()` returns the length required (in this case,
-the value returned by :code:`str.GetLenUsed()` will be greater than
-that returned by :code:`str.GetLenAlloc()` until corrected by
-application code.
+- If :code:`len_alloc` is not large enough, set :code:`len_used` to
+  the required length, and return with a return code of
+  :code:`C.YDB_ERR_INVSTRLEN`. In this case, :code:`len_used` will be
+  greater than :code:`len_alloc` until corrected by application code.
+- If :code:`str` has errors and is not in valid `zwrite format`_,, set
+  :code:`len_used` to zero, and return the error code returned by
+  `ydb_zwr2str_s()`_ prefixed by :code:`C.`, e.g.,
+  :code:`C.YDB_ERR_INVZWRITECHAR`.
+- Otherwise, set the buffer referenced by :code:`buf_addr` to the
+  unencoded string, set :code:`len_used` to the length, and return
+  with a return code of :code:`C.YDB_OK`.
 
 Note that the length of a string in `zwrite format`_ is always less
```

Comment:
Is this backwards? The original string is less than or equal to the zwrite format. The zwrite format can be much larger than the unencoded format.

Reply:
Yes it is. Fixed.